kernel/preemption/src/lib.rs

```diff
@@ -117,7 +118,9 @@ impl Drop for PreemptionGuard {
         assert!(
             self.cpu_id == cpu_id,
             "PreemptionGuard::drop(): BUG: CPU IDs did not match! \
-            This indicates an unexpected task migration across CPUs."
+            Task migrated from CPU {} to CPU {}.",
```

Comment:
nit: directly state it's not expected

```suggestion
            Task unexpectedly migrated from CPU {} to CPU {}.",
```